File: python/python/day7_oop.py
# Day 7 - Python OOP

# Base class
class Student:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def info(self):
        print(f"Student Name: {self.name}, Age: {self.age}")

# Derived class
class StudentMarks(Student):
    def __init__(self, name, age, marks):
        super().__init__(name, age)
        self.marks = marks

    def display(self):
        self.info()
        print(f"Marks: {self.marks}")

# Example usage
s1 = StudentMarks("Kalyani", 22, 78)
s1.display()

File: javascript/javascript/day7_oop.js
// Day 7 - JavaScript OOP

class Student {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }

    info() {
        console.log(`Student Name: ${this.name}, Age: ${this.age}`);
    }
}

class StudentMarks extends Student {
    constructor(name, age, marks) {
        super(name, age);
        this.marks = marks;
    }

    display() {
        this.info();
        console.log(`Marks: ${this.marks}`);
    }
}

// Example usage
let s1 = new StudentMarks("Kalyani", 22, 78);
s1.display();